x-pack/plugins/cases/public/components/case_view/case_view_page.tsx

```diff
@@ -146,6 +163,72 @@ export const CaseViewPage = React.memo<CaseViewPageProps>(
       }
     }, [onComponentInitialized]);
 
+    const tabs = useMemo(
+      () => [
+        {
+          id: CASE_VIEW_PAGE_TABS.ACTIVITY,
+          name: ACTIVITY_TAB,
+          content: (
+            <CaseViewActivity
+              getCaseUserActions={getCaseUserActions}
+              initLoadingData={initLoadingData}
+              ruleDetailsNavigation={ruleDetailsNavigation}
+              caseId={caseId}
+              caseData={caseData}
+              actionsNavigation={actionsNavigation}
+              showAlertDetails={showAlertDetails}
+              updateCase={updateCase}
+              fetchCaseMetrics={fetchCaseMetrics}
+              useFetchAlertData={useFetchAlertData}
+            />
+          ),
+        },
+        ...(!isMainApplication
+          ? [
+              {
+                id: CASE_VIEW_PAGE_TABS.ALERTS,
+                name: ALERTS_TAB,
+                content: (
+                  <EuiEmptyPrompt
+                    icon={<EuiLoadingLogo logo="logoKibana" size="xl" />}
+                    title={<h2>{'Alerts table placeholder'}</h2>}
+                  />
+                ),
+              },
+            ]
+          : []),
+      ],
+      [
+        actionsNavigation,
+        caseData,
+        caseId,
+        fetchCaseMetrics,
+        getCaseUserActions,
+        initLoadingData,
+        isMainApplication,
+        ruleDetailsNavigation,
+        showAlertDetails,
+        updateCase,
+        useFetchAlertData,
+      ]
+    );
+    const selectedTabContent = useMemo(() => {
+      return tabs.find((obj) => obj.id === activeTabId)?.content;
```

Comment:
Do you think is better to turn the `tabs` to a hash map instead of searching every time?

Reply:
It sounds reasonable but it is a minimal optimization really. JS engines are extremely fast on running loops and finds like this. In other plugins I've seen them using this same approach.

Reply:
Ok! Just wandering. No need to change it.